sa/storage-authority.go

```diff
@@ -277,6 +277,16 @@ func (ssa *SQLStorageAuthority) GetRegistration(id string) (reg core.Registratio
 	return
 }
 
+func (ssa *SQLStorageAuthority) GetRegistrationByKey(key jose.JsonWebKey) (reg core.Registration, err error) {
+	keyJson, err := key.MarshalJSON()
```

Comment:
I think the right thing here is actually json.Marshal(key). The MarshalJSON function is part of the interface JsonWebKey implements in order for the `encoding/json` package to know how to marshal it. I think I may have made the same mistake myself elsewhere in the codebase.